public/app/plugins/datasource/prometheus/language_utils.ts

```diff
@@ -1,8 +1,26 @@
 export const RATE_RANGES = ['1m', '5m', '10m', '30m', '1h'];
 
-export function processLabels(labels, withName = false) {
-  const values = {};
-  labels.forEach(l => {
+export const processHistogramLabels = (labels: string[]) => {
+  return labels.reduce(
+    (result, label) => {
+      const isHistogramValue = label.endsWith('_bucket') || label.endsWith('_bucket:sum_rate');
```

Comment:
While metrics (and histogram) naming is entirely up to the user I think we can catch most with a regex like `_bucket($|:)` which makes sure we hit metrics ending on bucket, and recording rules where the base name ends on bucket.

Reply:
Good catch, I've updated the code!